subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -2436,8 +2436,54 @@ ssize_t ztls_sendmsg_ctx(struct tls_context *ctx, const struct msghdr *msg,
 	ssize_t ret;
 	int i;
 
+	if (msg == NULL) {
+		errno = EINVAL;
+		return -1;
+	}
+
 	if (IS_ENABLED(CONFIG_NET_SOCKETS_ENABLE_DTLS) &&
 	    ctx->type == SOCK_DGRAM) {
+		if (CONFIG_NET_SOCKETS_DTLS_SENDMSG_BUF_SIZE > 0) {
+			static K_MUTEX_DEFINE(sendmsg_lock);
+			static uint8_t sendmsg_buf[CONFIG_NET_SOCKETS_DTLS_SENDMSG_BUF_SIZE];
```

Comment:
Implementation looks valid, thanks a lot!

I was just wondering if we could split that function into two separate, one that covers `if (CONFIG_NET_SOCKETS_DTLS_SENDMSG_BUF_SIZE > 0) {` case and the other one with code after `send_loop:` label. I think this would simplify reading and understanding this code by future contributors.

Reply:
The original function became quite bulky indeed. I've split it as suggested, one function for "merge and send" approach, and a second for "loop and send". Let me know if that's what you had in mind.

Reply:
Yes, that is exactly what I had in mind, looks great, thanks!